src/UI/Areas/Identity/Pages/Account/Manage/TwoFactorAuthentication.cshtml.cs

```diff
@@ -42,9 +43,22 @@ public class TwoFactorAuthenticationModel : PageModel
 
             HasAuthenticator = await _userManager.GetAuthenticatorKeyAsync(user) != null;
             Is2faEnabled = await _userManager.GetTwoFactorEnabledAsync(user);
+            IsMachineRemembered = await _signInManager.IsTwoFactorClientRememberedAsync(user);
             RecoveryCodesLeft = await _userManager.CountRecoveryCodesAsync(user);
 
             return Page();
         }
+
+        public async Task<IActionResult> OnPost()
+        {
+            var user = await _userManager.GetUserAsync(User);
+            if (user == null)
+            {
+                return NotFound($"Unable to load user with ID '{_userManager.GetUserId(User)}'.");
+            }
+
+            await _signInManager.ForgetTwoFactorClientAsync();
+            return RedirectToPage();
```

Comment:
Sorry, I should have included this in my review earlier. We should probably print a banner after the Redirect is complete that says the machine has been forgotten.

```C#
[TempData]
public string BannerMessage { get; set; }

...

await _signInManager.ForgetTwoFactorClientAsync();
BannerMessage = "The current browser has been forgotten. You will be asked to login with your credentials the next time you visit this website"
return RedirectToPage();
```

and display the banner in the page if it's set.

Reply:
This isn't forgetting the login cookie, note this is only clearing the two factor cookie, so it will be more that you will be prompted for two factor again in your next login.

Reply:
Might still want confirmation though to show people it works

Reply:
Yeah I just meant the banner text is wrong, we can show a banner, altho we aren't consistent about this right now I don't think

Reply:
And the banner should say "The current browser has been forgotten. When you login again from this browser you will be prompted for your 2fa code."

Reply:
@pranavkm what's the best pattern to follow for displaying the banner, do I need to go the full on Html.Partial/_statusMessage route that Manage/Index is currently doing?

Reply:
Actually I'll just use the existing statusMessage partial in here, @blowdart should we add status messages for all the buttons on the TFA page to be consistent?

Reply:
You could just do

```
// TwoFactorAuth.cshtml
if (Model.BannerMessage != null)
{
<div class="alert alert-success" role="alert">
  @Model.BannerMessage
</div>
}
```
if you set it via TempData. Shouldn't need a partial

Reply:
Ok sounds good.

Reply:
If we can be consistent we should :)

Reply:
I'm going to switch to using the StatusMessage consistently so all the manage views are consistently using the same partial and naming rather than mixing Banner and StatusMessage.  I'll also add some messages that will get shown at the top like

"Your authenticator app key has been reset, please reenable using the new key.";
"You have successfully added an authenticator."